### ISSUE TITLE
How to add resource identifier to the url

### ISSUE THREAD

**Issue by ecthelion77** [thomseddon/traefik-forward-auth#104]
Hello,

is it a way to add a variable to the url that is sent to the oidc provider ? 
I successfully use traefik-forward-auth with microsoft ADFS but i'm not able to use authorization from ADFS :

ADFS can be configured with two steps :

- first step : validate authentication with a callback identifier and authorize connection. This step is ok

url : ?client_id=****&redirect_uri=https%3A%2F%2Fservice.domain.tld%2F_oauth&response_type=code&scope=openid+profile+email&state=****%3Aoidc%3Ahttps%3A%2F%2Fservice.domain.tld%2F#/

- second step : allows to authorize (or not) for a specified group, mfa, ...
for that it's necessary to add a new variable in url :

?client_id=****&redirect_uri=https%3A%2F%2Fservice.domain.tld%2F_oauth&response_type=code&scope=openid+profile+email&state=****%3Aoidc%3Ahttps%3A%2F%2Fservice.domain.tld%2F#/&resource=<resource identifier>

In my case, I would add resource=service.domain.tld in the constructed url

**Comment by thomseddon** [comment]
There isn't currently a way to do this, it seems to be well defined: https://tools.ietf.org/html/draft-ietf-oauth-resource-indicators-08 so it makes sense to support his.

It could be supported with an additional "resource" configuration option under the provider, which could in turn pass this as an `AuthCodeOption`: https://godoc.org/golang.org/x/oauth2#Config.AuthCodeURL

**Comment by ecthelion77** [comment]
With your information, I'm trying to modify code to add this feature.

My first correct test :
in providers.go
````
// OAuthGetLoginURL provides a base "GetLoginURL" for proiders using OAauth2
func (p *OAuthProvider) OAuthGetLoginURL(redirectURI, state string) string {
	config := p.ConfigCopy(redirectURI)
	return config.AuthCodeURL(state, oauth2.SetAuthURLParam("resource", "testid"))
}
````
Now to be able to continue using your great tool properly I need to replace "testid" by variable that contains source web site.

For example:
my source : https://whoami.domain.tld
oauth proxy : https://openid.domain.tld
idp : https://sso.domain.tld/adfs

so current url is : 
https://sso.domain.tld/adfs/oauth2/authorize/?client_id=*****&redirect_uri=https%3A%2F%2Fopenid.domain.tld%2F_oauth&resource=testid&response_type=code&scope=openid+profile+email&state=*****%3Aoidc%3Ahttps%3A%2F%2Fwhoami.domain.tld%2F

as requested, resource is set.

Now I want replace resource=testid with resource=https%3A%2F%2Fwhoami.domain.tld%2F
but I don't find variable that contains this information.

It's possible for you to give me this information ?

**Comment by ecthelion77** [comment]
I have finally solved previous question !

My tests are ok, I added a new param pass-resource-id (bool) and if param set to true, url is posted to the IDP with resource=redirectBase.

I'm not able to push my branch to publish my modifications, can you allow me to push ?

**Comment by ecthelion77** [comment]
@thomseddon : have you read my previous post ?

**Comment by thomseddon** [comment]
Sorry I missed your posts! You will need to fork the repo and push to your own branch, could you send this as a PR? More info on how to do that is here: https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request

**Comment by thomseddon** [comment]
Merged pending release in 2.2
